lti_consumer/tests/unit/test_utils.py

```diff
@@ -0,0 +1,53 @@
+"""
+Utility functions used within unit tests
+"""
+
+from webob import Request
+from mock import Mock
+
+from xblock.fields import ScopeIds
+from xblock.runtime import KvsFieldData, DictKeyValueStore
+
+from workbench.runtime import WorkbenchRuntime
+
+
+def make_xblock(xblock_name, xblock_cls, attributes):
+    """
+    Helper to construct XBlock objects
+    """
+    runtime = WorkbenchRuntime()
+    key_store = DictKeyValueStore()
+    db_model = KvsFieldData(key_store)
+    ids = generate_scope_ids(runtime, xblock_name)
+    xblock = xblock_cls(runtime, db_model, scope_ids=ids)
+    xblock.category = Mock()
+    xblock.location = Mock(
+        html_id=Mock(return_value='sample_element_id'),
+    )
+    xblock.runtime = Mock(
+        hostname='localhost',
+    )
+    xblock.course_id = 'course-v1:edX+DemoX+Demo_Course'
+    for key, value in attributes.iteritems():
+        setattr(xblock, key, value)
```

Comment:
Actually, it looks like the XBlock object's __dict__ attribute is not updating it's attributes. I'm going to leave this as is for now.